src/wizards/newspack/views/settings/connections/webhooks/utils.tsx

```diff
@@ -81,3 +82,40 @@ export function getRequestStatusIcon(
 export function hasEndpointErrors( endpoint: Endpoint ): boolean {
 	return endpoint.requests.some( request => request.errors.length );
 }
+
+/**
+ * Validate endpoint URL.
+ *
+ * @param url The URL to validate.
+ * @return    Error message if URL is invalid, false otherwise.
+ */
+export function validateUrl( url: string ): string | false {
```

Comment:
This can be modernized to use [URL interface](https://developer.mozilla.org/en-US/docs/Web/API/URL) for example:
```ts
export function validateUrl( url: string ): string | false {
	if ( ! url ) {
		return __( 'URL is required.', 'newspack-plugin' );
	}
	try {
		const urlObject = new URL( url );
		// Ensure the protocol is either http or https
		if ( ! [ 'http:', 'https:' ].includes( urlObject.protocol ) ) {
			return __( 'URL must use HTTP or HTTPS protocol.', 'newspack-plugin' );
		}
		return false;
	} catch ( error ) {
		return __( 'Invalid URL format.', 'newspack-plugin' );
	}
}
```
There is also a function inside `@wordpress/url` called `isUrl` which also validates urls using the above pattern.

Reply:
Good call! Updated in 9eb25894d8850e5054d0ea0998e89d7eb8a9181e. While at it, I also restricted to HTTPS, which [is enforced in the backend](https://github.com/Automattic/newspack-plugin/blob/9eb25894d8850e5054d0ea0998e89d7eb8a9181e/includes/data-events/class-api.php#L248).